helix-core/src/test/java/org/apache/helix/task/TaskSynchronizedTestBase.java

```diff
@@ -181,12 +186,22 @@ protected void stopParticipants() {
 
   protected void stopParticipant(int i) {
     if (_participants.length <= i) {
-      throw new HelixException(
-          String.format("Can't stop participant %s, only %s participants" + "were set up.", i,
+      throw new HelixException(String
+          .format("Can't stop participant %s, only %s participants" + "were set up.", i,
               _participants.length));
     }
-    if (_participants[i] != null && _participants[i].isConnected()) {
-      _participants[i].syncStop();
+    if (_participants[i] != null) {
+      if (_participants[i].isConnected()) {
+        _participants[i].syncStop();
+      }
+      // Shutdown the state model factories to close all threads.
+      StateMachineEngine stateMachine = _participants[i].getStateMachineEngine();
+      if (stateMachine != null) {
+        StateModelFactory stateModelFactory = stateMachine.getStateModelFactory("Task");
```

Comment:
Nit: use the constant in TaskConstants instead of hard-coded string

Reply:
good point. Fixed